spec/src/main/asciidoc/core/packagingdeployment.asciidoc

```diff
@@ -198,32 +108,20 @@ For every type in the set of discovered types (as defined in <<type_discovery_st
-* if the class is an enabled bean, interceptor or decorator and if `ProcessBeanAttributes.veto()` wasn't called in previous step, fire an event which is a subtype of `ProcessBean`, as defined in <<process_bean>>.
-
-For each enabled bean, the container must search the class for producer methods and fields, as defined in <<producer_method>> and in <<producer_field>>, including resources, and for each producer:
+* determine which alternatives and interceptors are enabled, according to the rules defined in <<enablement>>.
 
-* if it is a producer method, fire an event of type `ProcessInjectionPoint` for each injection point in the method parameters, as defined in <<process_injection_point>>, and then
-* fire an event of type `ProcessProducer`, as defined in <<process_producer>>, and then
-* if the producer method or field is enabled, fire an event of type `ProcessBeanAttributes`, as defined in <<process_bean_attributes>>, and then
-* if the producer method or field is enabled and if `ProcessBeanAttributes.veto()` wasn't called in previous step, fire an event which is a subtype of `ProcessBean`, as defined in <<process_bean>>.
-
-For each enabled bean, the container must search for disposer methods as defined in <<disposer_method>>, and for each disposer method:
-
-* fire an event of type `ProcessInjectionPoint` for each injection point in the method parameters, as defined in <<process_injection_point>>.
-
-For each enabled bean, the container must search the class for observer methods, and for each observer method:
-
-* fire an event of type `ProcessInjectionPoint` for each injection point in the method parameters, as defined in <<process_injection_point>>, and then
-* fire an event of type `ProcessObserverMethod`, as defined in <<process_observer_method>>.
+For each enabled bean, the container must search the class for producer methods and fields, as defined in <<producer_method>> and in <<producer_field>>, including resources, and for disposer methods as defined in <<disposer_method>>, and for observer methods.
 
 Then, the container registers the `Bean` and `ObserverMethod` objects:
 
-* For each enabled bean that is not an interceptor or decorator, the container registers an instance of the `Bean` interface defined in <<bean>>.
+* For each enabled bean that is not an interceptor, the container registers an instance of the `Bean` interface defined in <<bean>>.
 * For each enabled interceptor, the container registers an instance of the `Interceptor` interface defined in <<interceptor>>.
-* For each enabled decorator, the container registers an instance of the `Decorator` interface defined in <<decorator>>.
 * For each observer method of every enabled bean, the container registers an instance of the `ObserverMethod` interface defined in <<observer_method>>.
 
+The container must execute all `@Registration` extension methods on build compatible extensions for each registered bean, interceptor, and observer method.
```

Comment:
```suggestion
The container must execute all `@Registration` extension methods in build compatible extensions for each registered bean, interceptor, and observer method.
```

Reply:
I am also not sure that you need to mention interceptors specifically since they are technically beans anyway.

Reply:
I'll remove this mention from the spec text, it's explicitly mentioned in the javadoc (and I believe that should stay).